log-collector/entrypoint.sh

```diff
@@ -30,6 +30,64 @@ MUST_GATHER_SCRIPT_URL="${MUST_GATHER_SCRIPT_URL:-https://raw.githubusercontent.
 ENABLE_GCP_SOS_COLLECTION="${ENABLE_GCP_SOS_COLLECTION:-false}"
 ENABLE_AWS_SOS_COLLECTION="${ENABLE_AWS_SOS_COLLECTION:-false}"
 
+# Mock mode for testing - auto-detect if running on KWOK fake nodes
+# KWOK nodes have the annotation kwok.x-k8s.io/node=fake
+MOCK_MODE="${MOCK_MODE:-auto}"
+MOCK_EXIT_CODE="${MOCK_EXIT_CODE:-0}"
+MOCK_SLEEP_DURATION="${MOCK_SLEEP_DURATION:-5}"
+
+# Auto-detect KWOK nodes for mock mode
+if [ "${MOCK_MODE}" = "auto" ]; then
+    # Check if this is a KWOK fake node
+    if kubectl get node "${NODE_NAME}" -o jsonpath='{.metadata.annotations.kwok\.x-k8s\.io/node}' 2>/dev/null | grep -q "fake"; then
+        MOCK_MODE="true"
+        echo "[INFO] Auto-detected KWOK fake node - enabling mock mode"
+    else
+        MOCK_MODE="false"
+    fi
+fi
```

Comment:
On second thought since this need kube permissions, let's drop auto and keep only true/false